### DISCUSSION TITLE
How to backup a JF instance ?

### DISCUSSION THREAD

**Discussion by nklamann** [jellyfin/jellyfin#7063] [Q&A] [answered]
Hello all,
i am starting with JF and installed it  (10.7.7 ) under debian 11.

I want to backup the configuration and metadata  and can accept a reasonable short downtime. The media files are cared of already.

My idea is 

- Stop the jellyfin service

backup the following files/dirs

- /etc/default/jellyfin

backup these subdirectories of $JELLYFIN_DATA_DIR

- data  
- metadata  
- plugins  
- root 

 (but NOT transcodes)

- Start the jellyfin service

Does that makle sense, am I missing something ?

Thanks for listening

Norbert

**Comment by aransou** [comment, marked as answer]
So far I understand this construct this should work. All files include the database are stored unter this directory jellyfin.

**Reply by gamerkonks** [reply]
Could you exclude the metadata directory and rebuild the metadata on restore?

**Comment by Abdull** [comment]
For Ubuntu 22.04 LTS with a Jellyfin Server `10.8.9` installation via [Jellyfin's apt repository](https://jellyfin.org/docs/general/installation/linux#ubuntu-repository), I have spotted the following *directories* of (possible) interest for backing up:

* `/var/lib/jellyfin` (database, metadata; also is Unix user `jellyfin`'s home directory)
* `/var/log/jellyfin` (logs)
* `/etc/jellyfin` (configuration)
* HTTP server access and error logs (e.g. from a [Apache2 HTTP Server reverse proxy setup](https://jellyfin.org/docs/general/networking/apache/))
* directories containing media files
* maybe `/usr/lib/jellyfin/bin/Resources/`
* any own custom Jellyfin-specific assets such as media library maintenance scripts and HTTP server configuration

Based on information gathered from:
* Jellyfin environment variables setup as per `systemctl status jellyfin` > `/lib/systemd/system/jellyfin.service` > `/etc/systemd/system/jellyfin.service.d/*` and `/etc/default/jellyfin`.
* https://jellyfin.org/docs/general/administration/configuration/
* https://www.reddit.com/r/jellyfin/comments/x8c3yj/full_backup_of_your_jellyfin_setup/inki0h6/
